### PR TITLE
Add .swiftlint.yml and fix warnings/errors

### PR DESCRIPTION
**Pull request by goranbrl**
# Description

Added .swiftlint.yml file and fixed resulting warnings/errors.

# Checklist:

- [x] Code is structured, well written using standard Swift coding style.
- [x] All public methods and properties have meaningful and concise documentation.
- [x] Used `public` modifier for all method and properties that could be changed from user of your feature, and `private` for internal properties.
- [x] Reduced the dependencies to minimum.

### CODE REVIEW

**Comment by goranbrl** [comment]
Bitrise is failing because Nimble has an issue with Xcode 12.5, it should pass after #42 is merged into this branch.

File: Catalog/Examples/Catalog/CatalogViewController.swift [hunk 1]
```diff
@@ -19,7 +19,7 @@ final class CatalogViewController: UIViewController {
     
     private lazy var _tableView: UITableView = _createTableView()
     
-    private lazy var _dataSourceDelegate: TableDataSourceDelegate = {
+    private lazy var _tableDataSource: TableDataSourceDelegate = {
```
Comment (z-turk3) [inline Catalog/Examples/Catalog/CatalogViewController.swift:22]: I see you removed `_` in the presenter above, do we need to remove them here as well?

File: Catalog/Examples/RxSwift/Paging/RxPagingViewController.swift [hunk 2]
```diff
@@ -23,7 +23,7 @@ final class RxPagingViewController: UIViewController, Refreshable {
 
     private let _disposeBag = DisposeBag()
 
-    private lazy var _dataSourceDelegate: TableDataSourceDelegate = {
+    private lazy var _tableDataSource: TableDataSourceDelegate = {
```
Comment (z-turk3) [inline Catalog/Examples/RxSwift/Paging/RxPagingViewController.swift:26]: In this VC also 😄

**Review by z-turk3 — COMMENTED** [review]
Left comment regarding `_` for private stuff, do we want to remove them all or not?

File: Catalog/Examples/Catalog/CatalogViewController.swift (hunk 1 above)
Comment (goranbrl) [reply Catalog/Examples/Catalog/CatalogViewController.swift:22]: For some reason swiftlint didn't scream here, but to align with the new coding style I'll remove them everywhere 😄

**Review by z-turk3 — APPROVED** [review]

**Pull request merged**